File: 백준/Silver/1059. 좋은 구간/좋은 구간.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int l = Integer.parseInt(br.readLine());
        String[] inputs = br.readLine().split(" ");
        int n = Integer.parseInt(br.readLine());

        List<Integer> s = new ArrayList<>();
        for(int i=0;i<l;i++){
            s.add(Integer.parseInt(inputs[i]));
        }
        if(s.contains(n)){
            System.out.println(0);
            return;
        }
        s.add(n);
        Collections.sort(s);
        int idx = s.indexOf(n);
        int left = (idx>0)?s.get(idx-1):0;
        int right = s.get(idx+1);
        System.out.println((n-left)*(right-n)-1);
    }
    
}

File: 백준/Silver/1059. 좋은 구간/좋은 구간.py
import sys
input = sys.stdin.readline

l = int(input())
s = list(map(int,input().split()))
n = int(input())
ans = []

if n in s:
    print(0)
else:
    s.append(n)
    s.sort()
    idx = s.index(n)
    left = s[idx-1] if idx>0 else 0
    right = s[idx+1]
    print((n-left)*(right-n)-1)

